File: src/main/java/com/ioannuwu/inline/domain/render/RangeHighlighterAdapter.java
package com.ioannuwu.inline.domain.render;

import com.intellij.openapi.Disposable;
import com.intellij.openapi.editor.markup.RangeHighlighter;

public class RangeHighlighterAdapter implements Disposable {

    private final RangeHighlighter rangeHighlighter;

    public RangeHighlighterAdapter(RangeHighlighter rangeHighlighter) {
        this.rangeHighlighter = rangeHighlighter;
    }

    @Override
    public void dispose() {
        rangeHighlighter.dispose();
    }
}

File: src/main/kotlin/com/ioannuwu/inline/domain/wrapper/RangeHighlighterAdapter.kt
package com.ioannuwu.inline.domain.wrapper

import com.intellij.openapi.Disposable
import com.intellij.openapi.editor.markup.RangeHighlighter

class RangeHighlighterAdapter(private val rangeHighlighter: RangeHighlighter): Disposable {
    override fun dispose() {
        rangeHighlighter.dispose()
    }

    override fun toString(): String {
        return "Adapter { $rangeHighlighter }"
    }
}